resdk/tests/functional/data_upload/e2e_upload.py

```diff
@@ -0,0 +1,268 @@
+                          'extract_protocol': 'Standard',
+                          'fragmentation_method': '',
+                          'growth_protocol': 'Standard media',
+                          'library_prep': 'Illumina',
+                          'treatment_protocol': 'Control'},
+            'reads_info': {'barcode': '', 'facility': '', 'instrument_type': ''}
+        }
+
+        # Check annotations
+        sample_1.update()
+        sample_2.update()
+        sample_2b.update()
+        sample_3.update()
+
+        self.assertEqual(sample_1.descriptor, ann_1)
+        self.assertEqual(sample_1.data[0].descriptor, reads_ann_1)
+        self.assertEqual(sample_1.tags, ['community:rna-seq'])
+        self.assertEqual(sample_2.descriptor, ann_2)
+        self.assertEqual(sample_2b.descriptor, ann_2)
+        self.assertEqual(sample_3.descriptor, ann_3)
+        self.assertEqual(sample_4.descriptor, {})
+        self.assertEqual(sample_5.descriptor, {})
+
+    def test_export(self):
+        resdk_logger.start_logging()
+
+        # Create the collection with named, unannotated samples
+        collection = self.res.collection.create(name='Test export annotation')
+
+        reads_1, reads_2 = self.get_reads(2, collection)
+
+        # Two different samples
+        sample_1 = reads_1.sample
+        sample_1.name = 'Sample 1'
+        sample_1.save()
+
+        sample_2 = reads_2.sample
+        sample_2.name = 'Sample 2'
+        ann_2 = {
+            'sample': {
+                'genotype': '',
+                'cell_type':
+                'Mixed',
+                'optional_char': [
+                    'LIBRARY_STRATEGY:Illumina Standard Prep ',
+                    'OTHER_CHAR_1:2x75 bp',
+                    'OTHER_CHAR_2:subdural cortical tumor, frontal/lateral'
+                    ' location. Easily isolated sample',
+                    'TISSUE:Tumor'
+                ],
+                'strain': 'N/A',
+                'source': 'Tumor',
+                'organism': 'Homo sapiens',
+                'molecule': 'total RNA',
+                'annotator': 'Tristan Brown',
+                'description': '',
+            }
+        }
+        sample_2.descriptor_schema = 'sample'
+        sample_2.descriptor = ann_2
+        sample_2.save()
+
+        reads_ann = {
+            'experiment_type': 'RNA-Seq',
+            'protocols': {
+                'growth_protocol': 'N/A',
+                'treatment_protocol': 'Control'
+            }
+        }
+        reads_2.descriptor_schema = 'reads'
+        reads_2.descriptor = reads_ann
+        reads_2.save()
+
+        # Export the new template
+        filepath = 'template1.xlsm'
+        try:
+            os.remove(filepath)
+        except OSError:
+            pass
+
+        collection.export_annotation(filepath)
+        assert os.path.exists(filepath)
+        # TODO: Include a file hash check
+        os.remove(filepath)
```

Comment:
Isn't this file automatically deleted after the test is completed?

Reply:
No. How would that happen with a file exported by ReSDK?